Improve nix flake

lluchs opened the pull request:
This fixes the nix flake (it didn't work with current nixpkgs since the old libdwarf was removed and some Python trouble) and implements some improvements:

- The plugin configuration can be overwritten by flake consumers and the flake can reduce the dependencies accordingly.
- PyPANDA is built separately (mostly to avoid rebuilding all of PANDA with every change while figuring out Python packaging)
- I also updated Wireshark to the newest version supported by the network plugin

cc @be32826 who built the flake originally

be32826 commented on flake.nix:
```diff
@@ -2,63 +2,83 @@
     };
+    # Override this input to change which plugins are built.
+    panda-config = {
+      url = "path:./panda/plugins/config.panda";
```
This will only work when running `nix build` in the cloned repo, but it won't work when used as a dependency. I get this error with `nix build github:lluchs/panda/flake`:

```
error: cannot fetch input 'path:./panda/plugins/config.panda' because it uses a relative path
```

Maybe plugin customization can be done at a different level from the flake inputs, such as with `.override`?

lluchs replied:
Ah, annoying. Thanks for catching that. `.override` unfortunately doesn't easily work without making the whole thing a nixpkgs overlay.

I've seen people suggest an input "path:/dev/null" for optional external configuration. I will try that.

be32826 replied:
> `.override` unfortunately doesn't easily work without making the whole thing a nixpkgs overlay.

I think you might be able to just change the panda package definition from

```nix
panda = pkgs.stdenv.mkDerivation {
  ...
}
```

to something like

```nix
panda = pkgs.callPackage
  ({ plugins ? builtins.readFile ./panda/plugins/config.panda }:
    pkgs.stdenv.mkDerivation {
      ...
    }) {}
```

lluchs commented:
I implemented `file:/dev/null` as input now since it was simpler from what I already had. `nix build github:lluchs/panda/flake` works now, as well as local builds from the repository and overriding the configuration from another flake.

lluchs commented:
I implemented `file:/dev/null` as input now since it was simpler from what I already had. `nix build github:lluchs/panda/flake` works now, as well as local builds from the repository and overriding the configuration from another flake.

lluchs commented:
I updated the PR with a better approach now. The flake now exposes a function `pkgsWithConfig` that allows setting `configFile` (as before) and `targetList`, for example:

```nix
    pandaPkgs = panda.packages.${system}.pkgsWithConfig {
      configFile = ./config.panda;
      targetList = [ "x86_64-softmmu" ];
    };
```

I also updated the lockfile (again) for current nixpkgs and libosi.
